File: RentABikeWTR_v1_7.Services/Database/Bicikli.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RentABikeWTR_v1_7.Services.Database
{
    public class Bicikli
    {
        // novi primjer[Key]
        public int BiciklId { get; set; }
        public string NazivBicikla { get; set; }
        public string Boja { get; set; }
        public decimal NabavnaCijena { get; set; }

        public DateTime GodinaProizvodnje { get; set; }
        public DateTime DatumOtpisa { get; set; }
        public decimal CijenaBicikla { get; set; } // ovo treba promijeniti


        public string VrstaRama { get; set; }

        public byte[]? Slika { get; set; }


        public int? VelicinaBiciklaID { get; set; }
        public virtual VelicineBicikla VelicinaBicikla { get; set; }
        public int? DrzavaID { get; set; }
        public virtual Drzave Drzava { get; set; }
        public int? StatusID { get; set; }// statusi, ako je iznajmljen, otpisan, NaStanju
        //Slobodan, Zauzet, Otpisan, Servis
        public virtual Statusi Status { get; set; }
        public int? PoslovnicaID { get; set; }
        public virtual Poslovnice Poslovnica { get; set; }

        public int? ProizvodjacBiciklaID { get; set; }
        public virtual ProizvodjaciBicikla ProizvodjacBicikla { get; set; }
        public int? TipBiciklaID { get; set; }
        public virtual TipoviBicikla TipBicikla { get; set; }
        public int? ModelBiciklaID { get; set; }
        public virtual ModeliBicikla ModelBicikla { get; set; }
        public virtual ICollection<Ocjene> Ocjene { get; set; }
        public virtual ICollection<Rezervacije> Rezervacije { get; set; }
        public virtual ICollection<Servisiranja> Servisiranje { get; set; }
    }
}

File: rentabikewtr_desktop/lib/model/bicikli.dart
import 'dart:ffi';

import 'package:json_annotation/json_annotation.dart';

part 'bicikli.g.dart';

@JsonSerializable()
class Bicikli {
  int? poslovnicaID;
  String? nazivBicikla;
  num? cijenaBicikla;
  String? boja;
  String? slika;
  int? modelBiciklaID;
  int? tipBiciklaID;
  int? proizvodjacBiciklaID;
  int? drzavaID;
  num? nabavnaCijena;
  String? vrstaRama;
  DateTime? godinaProizvodnje;
  int? statusID;
  int? velicinaBiciklaID;
  DateTime? datumOtpisa;

  Bicikli(
      {this.poslovnicaID,
      this.nazivBicikla,
      this.cijenaBicikla,
      this.boja,
      this.slika,
      this.modelBiciklaID,
      this.tipBiciklaID,
      this.proizvodjacBiciklaID,
      this.drzavaID,
      this.nabavnaCijena,
      this.vrstaRama,
      this.godinaProizvodnje,
      this.statusID,
      this.velicinaBiciklaID,
      this.datumOtpisa}) {}

  factory Bicikli.fromJson(Map<String, dynamic> json) =>
      _$BicikliFromJson(json);

  /// Connect the generated [_$BicikliToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$BicikliToJson(this);
}
